### ISSUE TITLE
Not possible to use RESP 3

### ISSUE THREAD

**Issue by MTigra** [RedisLabs/redisraft#100]
Looks like it is not possible to use redisraft with RESP 3
follower-proxy yes
7001 is nonleader node
```
$ redis-cli -p 7001 hello 3
(error) ERR bad reply from leader
```

**Comment by yossigo** [comment]
This needs to be fixed.
There are several things to consider here:
* `HELLO` shouldn't have been intercepted and relayed to the leader in the first place.
* The response handling part of the proxy should support RESP3.
* As part of proxy connection management (#35) we need to handle cases where we may have RESP2 and RESP3 sessions relayed concurrently. If we go for a dedicated connection per client that should be trivial.
*

**Comment by sjpotter** [comment]
first simple PR https://github.com/RedisLabs/redisraft/pull/169

**Comment by yossigo** [comment]
Fixed by #169
